### PR TITLE
New Feature: Allows custom group for showon

### PR DESCRIPTION
**Pull request by mvanvu**
Pull Request for Issue # .

### Summary of Changes
By default the showon feature detects the group from base fields. This PR comes to build for developer can specify the field which from the other group.

### Testing Instructions
Eg. I want to show the article title when the value of field **jform[attribs][show_title]** is show.

**Before PR:** no way to do (maybe, not sure)

**After PR**: Article title only show when the show title options is show.

Change the file **/administrator/components/com_content/models/forms/article.xml** line 21
**From**
```
<field 
			name="title" 
			type="text" 
			label="JGLOBAL_TITLE"
			description="JFIELD_TITLE_DESC"
			class="input-xxlarge input-large-text"
			size="40"
			required="true" 
		/>
```
**To**
```
<field 
			name="title" 
			type="text" 
			label="JGLOBAL_TITLE"
			description="JFIELD_TITLE_DESC"
			class="input-xxlarge input-large-text"
			size="40"
			required="true" 
                        showon="attribs.show_title:1"
		/>
```

The key changing here is **showon="attribs.show_title:1"**.
The custom group should only apply for the first time, for Eg: we need to expand the condition to show_title is empty OR show_title = 1.

`showon="attribs.show_title:[OR]show_title:1"
screenshot from 2017-12-01 22-49-26
screenshot from 2017-12-01 22-49-56

`

### CODE REVIEW

**Comment by GeraintEdwards** [comment]
I have tested this item :white_check_mark: successfully on 66dcdb2bab609e9fd5e674399f7a982d4c568089<br /><br />This works fine - a logical extension of showon.  I have been grappling with an issue like this for subforms so the idea could be quite useful

By the way you could use 

`showon="attribs.show_title:1,"`

Instead of the more confusing version you have.  It basically allows blank or 1 values to match<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18937">issues.joomla.org/tracker/joomla-cms/18937</a>.</sub>

**Comment by mvanvu** [comment]
Now the show on group is  improved overall.
For eg: content title showon `attribs.show_title:1[OR]images.float_intro:left`

File: libraries/src/Form/FormHelper.php [hunk 1]
```diff
@@ -480,8 +480,25 @@ public static function parseShowOnConditions($showOn, $formControl = null, $grou
...
 
+			if (strpos($showOnPartBlocks[0], '.') !== false)
+			{
+				if ($formControl)
+				{
+					$field = $formControl . ('[' . str_replace('.', '][', $showOnPartBlocks[0]) . ']');
+				}
+				else
+				{
+					$showOnParts = explode('.', $showOnPartBlocks[0]);
```
Comment (GeraintEdwards) [inline libraries/src/Form/FormHelper.php]: You are setting $showOnParts but this is an array your are running through at line 472 so this is not a good idea.

You should use a different variable name here

Comment (mvanvu) [reply libraries/src/Form/FormHelper.php]: Thanks for your review. This is my fault.

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 5a5eb72a0776e5043bc0a5781990fff03c500019

Test as decribed in first Comment.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18937">issues.joomla.org/tracker/joomla-cms/18937</a>.</sub>

**Comment by ghost** [comment]
@GeraintEdwards can you please retest?

**Comment by mvanvu** [comment]
@franz-wohlkoenig If there is not any problem I think it should be RTC.

**Comment by ghost** [comment]
@mvanvu i can't decide as there is only one successfully Test, Release Lead  can. So best is to get a second Test.

**Comment by grhcj** [comment]
I have not tested this item.<br /><br />Du you expect the title field to be bigger than the alias field?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18937">issues.joomla.org/tracker/joomla-cms/18937</a>.</sub>

**Comment by infograf768** [comment]
Changed PR title to New Feature

**Comment by coolcat-creations** [comment]
What is missing here for me is that this showon should also consider the global value. I have Global (Show) but the Alias is not shown.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18937">issues.joomla.org/tracker/joomla-cms/18937</a>.</sub>

**Comment by mvanvu** [comment]
@coolcat-creations as suggested from @GeraintEdwards just like this `showon="attribs.show_title:1,"`

**Comment by coolcat-creations** [comment]
I have tested this item :white_check_mark: successfully on 5a5eb72a0776e5043bc0a5781990fff03c500019<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18937">issues.joomla.org/tracker/joomla-cms/18937</a>.</sub>

**Comment by coolcat-creations** [comment]
Right, sorry - was not obvious to me that it reffered to global :)

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Pull request merged**